### PR TITLE
Fluter freezed v3.0.0

### PR DESCRIPTION
**Pull request by Parables**
Related # (issue)
https://github.com/dotansimha/graphql-code-generator/pull/8350#issuecomment-1305391678

### CODE REVIEW

**Comment by Parables** [comment]
Work is in progress on [this issue ](https://github.com/dotansimha/graphql-code-generator/issues/8330#issuecomment-1306629756) here

**Comment by denes16** [comment]
When will this be merged?

**Comment by Parables** [comment]
@charlypoly Why is the PR failing the tests? I am pretty sure all my tests pass. Could you look into that for me please?

File: packages/plugins/dart/flutter-freezed/src/index.ts [hunk 1]
```diff
@@ -1,38 +1,39 @@
 import { oldVisit, PluginFunction, Types } from '@graphql-codegen/plugin-helpers';
 import { transformSchemaAST } from '@graphql-codegen/schema-ast';
 import { GraphQLSchema } from 'graphql';
-import { FlutterFreezedPluginConfig } from './config.js';
-import { FreezedDeclarationBlock } from './freezed-declaration-blocks/index.js';
-import { schemaVisitor } from './schema-visitor.js';
-import { addFreezedImportStatements, DefaultFreezedPluginConfig } from './utils.js';
+import { FlutterFreezedPluginConfig } from './config';
+import { schemaVisitor } from './schema-visitor';
+import { buildImportStatements, defaultFreezedPluginConfig } from './utils';
```
Comment (charlypoly) [inline packages/plugins/dart/flutter-freezed/src/index.ts:6]: @Parables all relative imports should end with a `.js` (for ESM support)

This is what is causing the CI tests to fail

**Review by charlypoly — CHANGES_REQUESTED** [review]

File: packages/plugins/dart/flutter-freezed/src/index.ts (hunk 1 above)
Comment (Parables) [reply packages/plugins/dart/flutter-freezed/src/index.ts:6]: Thanks very much for the feedback

**Comment by Parables** [comment]
@denes16 @charlypoly @endigo @auaicn, I seriously need your feedback on these new breaking features.

Clone this [repo](https://github.com/Parables/graphql-codegen-flutter-freezed) where I do all my crazy experimentation before releasing them for this plugin

In an attempt to make the plugin more flexible for any use case, and provide more features to enhance the DevX of using GraphQL in Flutter projects, I have introduced some features but I fear I might be over-engineering something that nobody will ever use.

I have been working on this straight for a month now and I feel no one is going to ever need these features I am going sleepless nights working on.

**Comment by Parables** [comment]
I thought I was going crazy with these new features I was implementing but am I proud I did.

What's new and upcoming in this plugin is modular configuration using specific patterns.
You'll be trading the previous bloated singular configuration for a more compact(okay: maybe a little verbose), granular multi configuration.

Watch out for flutter-freezed-v4.0.0

@charlypoly I will need you to review this PR. 
Please let me know if you're available or if there's anyone I should reach out to

**Pull request closed without merging**